openedx-unsupported/configuration
# Decouple log syncing from AWS and OpenStack roles

This PR splits out the log syncing bits of the openstack and AWS roles to their own roles such that either or both can be enabled.

The aim is to decouple the log syncing destination from AWS/OpenStack so a server deployed on OpenStack can sync to S3 and a server deployed to AWS can sync to SWIFT, or a server can sync to both.

This replaces #3522 

**Sandbox URL**: 
- https://confpr4398.sandbox.opencraft.hosting/
- https://studio-confpr4398.sandbox.opencraft.hosting/

**Merge deadline**:  We would love to get this into Hawthorn

**Testing instructions**:
- Deploy a server using default log syncing settings
  - Log syncing should be disabled by default
  - Log rotation should still work
- Deploy a server with  ``COMMON_OBJECT_STORE_LOG_SYNC`` set, and AWS log sync settings configured
 - Logs should sync to S3 
- Deploy a server with ``COMMON_S3_LOG_SYNC`` set, and the AWS log sync settings configured
  - Logs should sync to S3
- Deploy a server with ``COMMON_SWIFT_LOG_SYNC`` set, and OpenStack settings configured. 
  - Logs should sync to SWIFT 
- Deploy a server with both `COMMON_S3_LOG_SYNC``, and ``COMMON_SWIFT_LOG_SYNC`` set, and with both AWS and OpenStack settings configured. 
  - Logs should sync to both S3 and SWIFT

**Reviewers**

**Settings**
```yaml
COMMON_SWIFT_LOG_SYNC: true
```
---

Make sure that the following steps are done before merging:
